docs/madr/decisions/014-mesh-circuit-breaker-policy.md

```diff
@@ -0,0 +1,252 @@
+inbound clusters, and with new policy matching it may be so simple to just allow to specify`from.targetRef` section
+
+Additional changes to the original `CircuitBreaker` policy consists of:
+
+- changing word `Errors` to `Failures` in detectors
+- adding `origin` suffix to the local failures detector
+
+It means `totalErrors`, `gatewayErrors` and `localErrors`becomes: `totalFailures`, `gatewayFailures`
+and `localOriginFailures` respectively.
+
+### Current configuration
+
+Below is the sample `MeshCircuitBreaker` configuration
+
+```yaml
+conf:
+  interval: 1s
+  baseEjectionTime: 30s
+  maxEjectionPercent: 20
+  splitExternalAndLocalErrors: false
+  thresholds:
+    maxConnections: 2
+    maxPendingRequests: 2
+    maxRequests: 2
+    maxRetries: 2
+  detectors:
+    totalErrors:
+      consecutive: 20
+    gatewayErrors:
+      consecutive: 10
+    localErrors:
+      consecutive: 7
+    standardDeviation:
+      requestVolume: 10
+      minimumHosts: 5
+      factor: 1.9
+    failure:
+      requestVolume: 10
+      minimumHosts: 5
+      threshold: 85
+
+```
+
+### Specification
+
+#### Top level
+
+As a part of this MADR root `targetRef`'s kind can be configured as one
+of `Mesh|MeshSubset|MeshService|MeshServiceSubset`. When we finally introduce `MeshHTTPRoute` kind and if there will be
+a reasonable use case, in the next iteration `MeshGatewayRoute` and `MeshHTTPRoute` could be added as a viable kinds.
+
+```yaml
+targetRef:
+  kind: Mesh|MeshSubset|MeshService|MeshServiceSubset
+  name: ...
+```
+
+#### From level
+
+```yaml
+to:
+- targetRef:
+    kind: Mesh
+    name: ...
+```
+
+#### To level
+
+```yaml
+to:
+- targetRef:
+    kind: Mesh|MeshService
+    name: ...
+```
+
+#### Minimal configuration
+
+As all the connection pool related properties have default values, there is no need to specify any of these to apply the
+policy
+
+```yaml
+default:
+  connectionPool: { }
+```
+
+### Examples
+
+#### Default configuration
+
+##### `MeshCircuitBreaker`
+
+```yaml
+type: MeshCircuitBreaker
+mesh: default
+name: default-circuit-breaker
+spec:
+  targetRef:
+    kind: Mesh
+    name: default
+  from:
+  - targetRef:
+      kind: Mesh
+      name: default
+    default:
+      disabled: false
+      connectionPool:
+        maxConnections: 1024
+        maxPendingRequests: 1024
+        maxRetries: 3
+        maxRequests: 1024
+  to:
+  - targetRef:
+      kind: Mesh
+      name: default
+    default:
+      disabled: false
+      connectionPool:
+        maxConnections: 1024
+        maxPendingRequests: 1024
+        maxRetries: 3
+        maxRequests: 1024
+```
+
+#### Extensive configuration
+
+```yaml
+type: MeshCircuitBreaker
+mesh: default
+name: extensive-circuit-breaker
+spec:
+  targetRef:
+    kind: Mesh
+    name: default
+  from:
+  - targetRef:
+      kind: Mesh
+      name: default
+    default:
+      disabled: true
```

Comment:
I don't think you can disable connection pool limits. Should this be in outlier detection?

Reply:
That's the case, and here is the discussion I started about it: https://github.com/kumahq/kuma/pull/5379#discussion_r1037107808